src/__tests__/api/updateContactData.test.ts

```diff
@@ -0,0 +1,176 @@
+import { beforeAll, describe, expect, it } from "vitest";
+import { prisma } from "@/lib/prisma";
+import { hashPassword } from "@/lib/hashPassword";
+import {
+  createMocks,
+  createRequest,
+  createResponse,
+  RequestMethod,
+} from "node-mocks-http";
+import { NextApiRequest, NextApiResponse } from "next";
+import { ShirtSize } from "@prisma/client";
+import { ShirtStyles } from "@/types/shirt";
+import updateContactDataHandler from "@/pages/api/user/updateContactData";
+type ApiRequest = NextApiRequest & ReturnType<typeof createRequest>;
+type APiResponse = NextApiResponse & ReturnType<typeof createResponse>;
+
+const dummyEmail = "upsertUser@test.com";
+
+const mailingAddressDB = {
+  street: "Calle",
+  externalNumber: "#8Bis",
+  zipcode: "01234",
+  country: "MEX",
+  state: "Aguascalientes",
+  references: "Por ahí",
+  phone: "5511223344",
+  county: "Aguascalientes",
+  neighborhood: "Aguascalientes",
+  name: "Yosshua V",
+};
+
+const validUserInput = {
+  firstName: "Yosshua",
+  lastName: "Villasana",
+  preferredName: "Yosshua",
+  birthDate: new Date("2006-11-24").toISOString(),
+  pronouns: "HE",
+  governmentId: "HEGG061124MVZRRL02",
+  shirtSize: ShirtSize.M,
+  shirtStyle: ShirtStyles[0],
+};
+
+beforeAll(async () => {
+  // Upsert the valid user Auth
+  await prisma.userAuth.upsert({
+    where: { email: dummyEmail },
+    update: {},
+    create: { email: dummyEmail, password: hashPassword("pass") },
+  });
+
+  const authUser = await prisma.userAuth.findUnique({
+    where: { email: dummyEmail },
+  });
```

Comment:
```suggestion
  const authUser = await prisma.userAuth.upsert({
    where: { email: dummyEmail },
    update: {},
    create: { email: dummyEmail, password: hashPassword("pass") },
  });
```

El prisma `upsert` ya te devuelve el authUser que quieres usar